tests/data/semparse/worlds/world_test.py

```diff
@@ -89,6 +89,14 @@ def test_get_paths_to_root_with_recursion(self):
     # we'll mostly use the NLVR world to test them, as it's a simpler world than the WikiTables
     # world.
 
+    def test_get_action_sequence_removes_currying(self):
+        world = self.wikitables_world
+        logical_form = ("(argmax (number 1) (number 1) (fb:row.row.division fb:cell.2) "
+                        "(reverse (lambda x ((reverse fb:row.row.index) (var x))))")
+        parsed_logical_form = world.parse_logical_form(logical_form)
+        action_sequence = world.get_action_sequence(parsed_logical_form)
+        assert 'r -> [<d,<d,<#1,<<d,#1>,#1>>>>, d, d, r, <d,r>]' in action_sequence
```

Comment:
Don't you also want to show that the functions are uncurried in their signatures? May be something along the lines of  `r -> [<(d,d,#1,<d,#1>),#1>, d, d, r, <d,r>]`? This is easier to read, and also accurate, since `<d,<d,<#1,<<d,#1>,#1>>>>` still implies currying (a function that takes `d` and returns `<d,<#1,<<d,#1>,#1>>>`).

Reply:
That's not what PNP does.  I don't have a strong opinion here, except doing that means changing a whole lot of things in the type declaration, I think.

Reply:
If it involves changing many more things than you are currently, it's probably not worth it. I just think this is cleaner, and I though I'd bring it up since you're changing things anyway. Also, are you completely removing currying, and not leaving it as an option? Looks like you are. If you have both curried and uncurried functions, then this point becomes more important.